### ISSUE TITLE
"max words" confusion and zero loss with alpaca dataset

### ISSUE THREAD

**Issue by iocuydi** [facebookresearch/llama-recipes#69]
I am trying to finetune LLAMA-2 7B using another instruction tuning dataset. It is pretty similar in format to the alpaca dataset, but the instructions are slightly longer. 

I'm often getting zero loss when training the model, and after some debugging i realized the the labels tensor used to evaluate the loss is usually full of zeros. I traced this back to the "max words" parameter in the alpaca dataset. https://github.com/facebookresearch/llama-recipes/blob/905f633dab92688b0a989f8d5cd11d86f882f534/ft_datasets/alpaca_dataset.py#L29

With a sufficiently long instruction, lines 63-65 beginning here https://github.com/facebookresearch/llama-recipes/blob/905f633dab92688b0a989f8d5cd11d86f882f534/ft_datasets/alpaca_dataset.py#L63 end up padding away the entire labels and replacing it with zeros, because the raw prompt is longer than the "example" (after the example is adjusted to fit max_words).

When I change max_words to a larger value, I get a lot of cuda errors like this:
read: [59,0,0] Assertion `srcIndex < srcSelectDimSize` failed.                                                                         
/opt/conda/conda-bld/pytorch_1682343967769/work/aten/src/ATen/native/cuda/Indexing.cu:1146: indexSelectLargeIndex: block: [623,0,0], th

This even happens with QLora occupying only 10gb of an 80gb GPU so i don't think it is a memory issue? 
Is there a way I can do instruction tuning with somewhat longer instructions? On some investigation, even a significant portion of the Alpaca dataset itself seems to get padded away to zero by this....

**Comment by chauhang** [comment]
@iocuydi Can you please share the full output logs with errors? It will also help if you can share the code changes that you made for the custom dataset

**Comment by iocuydi** [comment]
After further debugging I realized I had forgotten to wrap the call to my dataset in "partial" as done here with alpaca:
https://github.com/facebookresearch/llama-recipes/blob/2243b962fa98c9c344f8f1a62be771bb1b71c56f/utils/dataset_utils.py#L17

After making this adjustment, I was able to pass much larger values of "max_words" without issue, and upon doing so I saw training behave as expected, with no more zero loss.

**Comment by timothylimyl** [comment]
@chauhang why do we name it `max_words`?  isn't it max_tokens?

**Comment by timothylimyl** [comment]
another part that is confusing to me, why is the max_words just set to 224?
